_source/_posts/2019-09-18-build-a-simple-dotnet-core-app-in-docker.md

```diff
@@ -0,0 +1,343 @@
+
+FROM mcr.microsoft.com/dotnet/core/aspnet:2.2 AS base
+WORKDIR /app
+EXPOSE 5001
+
+FROM base AS final
+WORKDIR /app
+COPY --from=publish /app .
+ENTRYPOINT ["dotnet", "OktaMvcLogin.dll"]
+```
+
+You will see a few `FROM` commands, each with an `AS` clause. This syntax provides multi-stage builds, the key to getting rid of unnecessary dependencies. In plain English, your build process is now:
+
+1. Use the SDK image to create a release build of the application. Call this stage 'build'
+2. Use the 'build' stage image to publish the application to the 'app' folder. Call this stage 'publish'
+3. Download the Microsoft-provided ASP.NET core image, which has only runtime components. Call this stage 'base'
+4. Using the 'base' stage image, copy the contents of the 'app' directory from the 'publish' stage. Call this stage 'final'
+
+So, your `Dockerfile` now uses the SDK image to build your application, then discards that image and uses a runtime image to run the application.
+
+## Run the ASP.NET Core Application in Docker
+
+The `ENTRYPOINT` command merits special attention. So far you've seen how Dockerfiles define a series of filesystem transformations, but more often than not, a Docker container is executable. By that, I mean that you run the container in Docker, and the result is a fully configured, running application. `ENTRYPOINT` is one of the mechanisms that make that work. When you run a container, Docker executes the command specified by the `ENTRYPOINT`. In the case of your application, that command is `dotnet OktaMVCLogin.dll`.
+
+So now...
+
+```bash
+docker build -t oktamvclogin .
+docker run oktamvclogin
+```
+
+... throws the same exception as before:
+
+```bash
+Unhandled Exception: System.ArgumentNullException: Your Okta URL is missing. Okta URLs should look like: https://{yourOktaDomain}. You can copy your domain from the Okta Developer Console.
+```
+
+Only this time, it's Dockerized. How's that for progress?
+
+The application doesn't work because you removed the sensitive configuration from `appsettings.json`.
+
+## Pass Configuration to Docker
+
+To fix this problem, we need to pass configuration to the Docker container as environment variables. ASP.NET Core picks up all environment variables prefixed with `ASPNETCORE_` and converts `__` into `:`. To pass the configuration values for `Okta:ClientId`, `Okta:ClientSecret` and `Okta:Domain` modify your command like this:
+
+```bash
+docker run -e ASPNETCORE_Okta__ClientId="{yourClientId}" \
+-e ASPNETCORE_Okta__ClientSecret="{yourClientSecret}" \
+-e ASPNETCORE_Okta__Domain="https://{yourOktaDomain}" \
+oktamvclogin
+```
+
+This time the result will be a bit healthier:
+
+```bash
+Hosting environment: Production
+Content root path: /app
+Now listening on: 'http://[::]:80'
+Application started. Press Ctrl+C to shut down.
+```
+
+>**Note**: you may also see a 'No XML Encryptor' warning. You can ignore that for this walkthrough.
```

Comment:
I would remove the > to make this a blockquote and change the note to be all caps. **NOTE:**